python/src/com/jetbrains/python/codeInsight/intentions/PyInvertIfConditionIntention.kt

```diff
@@ -0,0 +1,341 @@
+// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.jetbrains.python.codeInsight.intentions
+
+import com.intellij.codeInsight.controlflow.ControlFlowUtil
+import com.intellij.codeInspection.SuppressionUtil
+import com.intellij.lang.ASTNode
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.*
+import com.intellij.psi.impl.source.codeStyle.CodeEditUtil
+import com.intellij.psi.tree.TokenSet
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.util.parents
+import com.intellij.psi.util.parentsOfType
+import com.intellij.psi.util.siblings
+import com.intellij.refactoring.suggested.endOffset
+import com.intellij.refactoring.suggested.startOffset
+import com.intellij.util.IncorrectOperationException
+import com.jetbrains.python.PyBundle
+import com.jetbrains.python.PyTokenTypes
+import com.jetbrains.python.codeInsight.controlflow.ControlFlowCache
+import com.jetbrains.python.codeInsight.controlflow.ReadWriteInstruction
+import com.jetbrains.python.codeInsight.controlflow.ScopeOwner
+import com.jetbrains.python.psi.*
+import com.jetbrains.python.psi.impl.PyPsiUtils
+import com.jetbrains.python.refactoring.ConditionUtil
+
+/**
+ * Inverts 'if' condition branches
+ *
+ * @author Vasya Aksyonov
+ */
+class PyInvertIfConditionIntention : PyBaseIntentionAction() {
+  private companion object {
+    val insignificantTokenSet = TokenSet.create(TokenType.WHITE_SPACE, PyTokenTypes.END_OF_LINE_COMMENT)
+    const val PYLINT_COMMENT_PREFIX = "# pylint:"
+  }
+
+  init {
+    text = PyBundle.message("INTN.invert.if.condition")
+  }
+
+  override fun getFamilyName(): String = text
+
+  override fun isAvailable(project: Project, editor: Editor, file: PsiFile): Boolean {
+    if (file !is PyFile) {
+      return false
+    }
+
+    val element = file.findElementAt(editor.caretModel.offset)!!
```

Comment:
This throws a `NullPointerException` when placing the caret in the last line of a file. It would be better to check for null.

Reply:
Great, thanks, will fix that and check other similar issues too! Just didn't know how to cause such a behavior :D

Reply:
Fixed